src/Generator.php

```diff
@@ -75,9 +76,10 @@
     /**
      * {@inheritdoc}
      */
-    public function init()
+    public function __construct()
     {
-        parent::init();
+
+        //parent::__construct();
```

Comment:
I've tried to implement `Initiable`, but the init() method wasn't called.
What's the good way to do things here ?

Reply:
Problem with moving everything to constructor is that you just don't have `$this->templates` yet in constructor.
So, templates should be passed to constructor or initialization should be moved to init.